scripts/test.bash

```diff
@@ -0,0 +1,22 @@
+ #!/bin/bash
+(set -o igncr) 2>/dev/null && set -o igncr; # For Cygwin/MSys2 on Windows compatibility
+
+cargo build
+if [ $? != 0 ]; then
+    exit 1
+fi
+cargo test
+if [ $? != 0 ]; then
+    exit 1
+fi
+
+cd xray_electron
+npm install
+if [ $? != 0 ]; then
+    exit 1
+fi
+npm test
+if [ $? != 0 ]; then
+    exit 1
+fi
```

Comment:
I think you could avoid a lot of these `if` statements by adding `set -e` to the top of the script.

Reply:
```set -e``` introduces a fair few hidden footguns by aborting after any non-zero exit code. If I remember correctly, there's also some differing behaviour between ```set -e``` in GNU Bash, MSys2 and Cygwin. I would argue against it, and personally prefer the more manual error handling. However, I can change it.

Reply:
:+1: No nevermind, this seems fine.